Cython/Utility/ModuleSetupCode.c

```diff
@@ -1019,14 +1102,21 @@ static CYTHON_SMALL_CODE PyObject* ${pymodule_create_func_cname}(PyObject *spec,
     Py_DECREF(modname);
     if (unlikely(!module)) goto bad;
 
-    moddict = PyModule_GetDict(module);
+#if CYTHON_COMPILING_IN_LIMITED_API
+    if (unlikely(PyModule_AddStringConstant(module, "loader", "__loader__") < 0)) goto bad;
+    if (unlikely(PyModule_AddStringConstant(module, "origin", "__file__") < 0)) goto bad;
+    if (unlikely(PyModule_AddStringConstant(module, "parent", "__package__") < 0)) goto bad;
+    if (unlikely(PyModule_AddStringConstant(module, "submodule_search_locations", "__path__") < 0)) goto bad;
+#else
+    PyObject *moddict = PyModule_GetDict(module);
```

Comment:
We still try to generate C89 code (thank you, good old MSVC!), so this declaration must stay at the beginning of the function. Although, we might actually reconsider that decision before this PR gets merged and released. If that's in Cython 3.1, then we'd require CPython 3.5+, in which case C99 might be a reasonable thing to depend on as well.

Anyway, since it's probably easy to keep up C89 compatibility in this PR, we should stick to it for now.

Reply:
Still TODO, will come back to this

Reply:
Done!